planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp

```diff
@@ -2841,6 +2864,52 @@ void AvoidanceModule::modifyPathVelocityToPreventAccelerationOnAvoidance(Shifted
     parameters_->max_avoidance_acceleration, vmax, ego_idx, target_idx);
 }
 
+PathWithLaneId AvoidanceModule::extendBackwardLength(const PathWithLaneId & original_path) const
+{
+  // special for avoidance: take behind distance upt ot shift-start-point if it exist.
+  const auto longest_dist_to_shift_point = [&]() {
+    double max_dist = 0.0;
+    for (const auto & pnt : path_shifter_.getShiftLines()) {
+      max_dist = std::max(max_dist, calcDistance2d(getEgoPose(), pnt.start));
+    }
+    for (const auto & sp : registered_raw_shift_lines_) {
+      max_dist = std::max(max_dist, calcDistance2d(getEgoPose(), sp.start));
+    }
+    return max_dist;
+  }();
+
+  const auto extra_margin = 10.0;  // Since distance does not consider arclength, but just line.
```

Comment:
should this be `constexpr` ? Or it might be parameterized in the future?

Reply:
I'll fix and use `constexpr`.

Reply:
fix in https://github.com/autowarefoundation/autoware.universe/pull/2987/commits/da1273324ee3cb8f8da1c187bc4f26a680f37034.